crates/wast/src/component/types.rs

```diff
@@ -800,7 +807,7 @@ impl<'a> Parse<'a> for ComponentTypeDecl<'a> {
         } else if l.peek::<kw::r#type>() {
             Ok(Self::Type(parser.parse()?))
         } else if l.peek::<kw::alias>() {
-            Ok(Self::Alias(Alias::parse_outer_type_alias(parser, false)?))
+            Ok(Self::Alias(parser.parse()?))
```

Comment:
More justification for improving the error message in the validator if we're also relaxing the parsing in `wat` in component/instance type declarations.